cms/src/main/java/uk/nhs/digital/externalstorage/s3/S3ObjectDescriber.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.digital.externalstorage.s3;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import javax.xml.bind.DatatypeConverter;
+
+public class S3ObjectDescriber {
+
+    MessageDigest md = null;
+    String bucketName;
+
+    public S3ObjectDescriber(String bucketName) {
+        this.bucketName = bucketName;
+
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    public String generateObjectKey(String fileName) {
+        String fileNameSalt = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(new Date());
```

Comment:
It feels like it should be enough to just use `String fileNameSalt = UUID.randomUUID().toString();`